backfill/2020-03-19-replay-payload-bytes-decoded/README.md

```diff
@@ -0,0 +1,152 @@
+# Backfill from `payload_bytes_decoded` to stable tables
+
+We had to kill a BQ live sink job on 2020-03-17 due to it getting into a bad
+state. See https://bugzilla.mozilla.org/show_bug.cgi?id=1622977 for details.
+Some portion of records (~20 GB of compressed JSON data) did not get loaded
+to live tables in the telemetry namespace.
+
+We choose the backfill-17 project for doing this work to match the date (17th).
+
+## Stage target payloads into a table
+
+First, we stage a backfill table following the pbd structure and we run a
+query to populate it:
+
+```
+# Create the destination dataset
+bq mk moz-fx-data-backfill-17:payload_bytes_decoded
+
+# Copy a small pbd table (we choose core_v1) to the backfill project;
+# we're going to overwrite the data, all we care about is getting the right structure.
+bq cp moz-fx-data-shared-prod:payload_bytes_decoded.telemetry_telemetry__core_v1 moz-fx-data-backfill-17:payload_bytes_decoded.telemetry_all
+
+# Now we run our query to populate the table.
+bq query --nouse_legacy_sql --project_id=moz-fx-data-backfill-17 --destination_table moz-fx-data-backfill-17:payload_bytes_decoded.telemetry_all --replace -n 0 < query.sql
+```
+
+The query handles deduplication, so we'll be able to copy the results directly
+into place in the historical ping tables (`_stable` tables). The query took
+about 30 minutes to complete and scanned about 5 TB of compressed data.
+
+## Run a Dataflow job to populate temporary stable tables
+
+We launch the Dataflow job:
+
```

Comment:
In order to get this to work I had to create service accounts keys for the backfill project and set them as `GOOGLE_APPLICATION_CREDENTIALS`. Just running `gcloud auth application-default login` didn't work.